### PR TITLE
Fix/bookings unintended confirmations on rejection 6975

### LINKED ISSUE #6975: bookings.tsx: unintended confirmations on rejection
https://github.com/calcom/cal.com/blob/156f3e5de2697c4dbb7abfbdf2d6f143e19cd9fe/packages/trpc/server/routers/viewer/bookings.tsx#L778

```javascript


    /** When a booking that requires payment its being confirmed but doesn't have any payment,
     * we shouldn’t save it on DestinationCalendars
     *
     * FIXME: This can cause unintended confirmations on rejection.
     */
    if (booking.payment.length > 0 && !booking.paid) {
      await prisma.booking.update({

```

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

There was some cases when trying to reject booking will result in confirming it instead.
Fixes #6975 

- Added redirection to payment page if booking was rejected/cancelled with/without payment.

https://www.loom.com/share/b65a2a4e837449e7ba66023727794724

**Environment**: Staging(main branch)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Setup a booking that requires confirmation or payment.
- [ ] Try to reject it and error shouldn't display anymore.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/pages/booking/[uid].tsx [hunk 1]
```diff
@@ -208,7 +208,7 @@ export default function Success(props: SuccessProps) {
   }
   const status = props.bookingInfo?.status;
   const reschedule = props.bookingInfo.status === BookingStatus.ACCEPTED;
-  const cancellationReason = props.bookingInfo.cancellationReason;
+  const cancellationReason = props.bookingInfo.cancellationReason || props.bookingInfo.rejectionReason;
```
Comment (alannnc) [inline apps/web/pages/booking/[uid].tsx:211]: With this we should display either message.

File: apps/web/pages/booking/[uid].tsx [hunk 2]
```diff
@@ -440,6 +440,17 @@ export default function Success(props: SuccessProps) {
...
+                  {props.paymentStatus && (
+                    <h4>
+                      {!props.paymentStatus.success &&
+                        !props.paymentStatus.refunded &&
+                        t("booking_with_payment_cancelled")}
+                      {props.paymentStatus.success &&
+                        !props.paymentStatus.refunded &&
+                        t("booking_with_payment_cancelled_already_paid")}
+                      {props.paymentStatus.refunded && t("booking_with_payment_cancelled_refunded")}
+                    </h4>
```
Comment (alannnc) [inline apps/web/pages/booking/[uid].tsx:452]: An extra step in booking page to let the user know paymentStatus.

File: apps/web/pages/booking/[uid].tsx [hunk 3]
```diff
@@ -1115,6 +1127,20 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
...
+    select: {
+      success: true,
+      refunded: true,
+    },
+  });
+  const paymentStatus = {
+    success: payment?.success,
+    refunded: payment?.refunded,
+  };
+
```
Comment (alannnc) [inline apps/web/pages/booking/[uid].tsx:1143]: Fetching payment but don't sending it via props.

File: packages/features/ee/payments/pages/payment.tsx [hunk 4]
```diff
@@ -104,6 +109,19 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
...
+      booking.status as BookingStatus
+    )
+  ) {
+    return {
+      redirect: {
+        destination: `/booking/${booking.uid}`,
+        permanent: false,
+      },
+    };
+  }
```
Comment (alannnc) [inline packages/features/ee/payments/pages/payment.tsx:123]: Redirect to booking page so we can display booking/event as cancelled.

File: packages/trpc/server/routers/viewer/bookings.tsx [hunk 5]
```diff
@@ -772,12 +772,8 @@ export const bookingsRouter = router({
...
     if (isConfirmed) throw new TRPCError({ code: "BAD_REQUEST", message: "Booking already confirmed" });
 
-    /** When a booking that requires payment its being confirmed but doesn't have any payment,
-     * we shouldn’t save it on DestinationCalendars
-     *
-     * FIXME: This can cause unintended confirmations on rejection.
-     */
-    if (booking.payment.length > 0 && !booking.paid) {
+    // If booking requires payment and is not paid, we don't allow confirmation
+    if (confirmed && booking.payment.length > 0 && !booking.paid) {
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer/bookings.tsx:776]: It looks like previous code was not using confirmed input and now we using it to handle actions properly.

File: packages/trpc/server/routers/viewer/bookings.tsx [hunk 6]
```diff
@@ -1083,66 +1080,66 @@ export const bookingsRouter = router({
...
 
-          // Posible to refactor TODO:
-          const paymentApp = appStore[paymentAppCredential?.app?.dirName as keyof typeof appStore];
-          if (!(paymentApp && "lib" in paymentApp && "PaymentService" in paymentApp.lib)) {
-            console.warn(`payment App service of type ${paymentApp} is not implemented`);
-            return null;
-          }
+            // Posible to refactor TODO:
+            const paymentApp = appStore[paymentAppCredential?.app?.dirName as keyof typeof appStore];
+            if (!(paymentApp && "lib" in paymentApp && "PaymentService" in paymentApp.lib)) {
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer/bookings.tsx:1132]: indentation because conditional

File: apps/web/pages/booking/[uid].tsx (hunk 3 above)
Comment (zomars) [reply apps/web/pages/booking/[uid].tsx:1143]: It is being sent via props in paymentStatus. I don't see the difference.

**Review by emrysal — APPROVED** [review]
Confused by the paymentStatus, but other than that looks good to me, less hard crashes 👍

**Pull request merged**